File: src/main/java/com/testcoders/Product.java
package com.testcoders;

public class Product {
    private final String name;
    private int barCode;
    private final int discount;
    private double price;

    public Product(String name, int barCode, double price, ProductCategory productCategory) {
        this.name = name;
        this.barCode = barCode;
        this.price = price;
        this.discount = productCategory.getDiscountValue();
    }

    public int getBarCode() {
        return barCode;
    }

    public void setBarCode(int barCode) {
        this.barCode = barCode;
    }

    public double getPrice() {
        double discountPrice = price * (100-this.discount) / 100;
        return  Math.rint(discountPrice * 100) / 100;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}

File: SuperMarketApp/Service/Models/Product.cs
using Service.Enum;

namespace Service.Models
{
    public class Product
    {
        public string Name { get; set; }
        public int Barcode { get; set; }
        public double Price { get; set; }
        public Discount Discount { get; set; }

        public Product(string name, int barcode, double price, Discount discount = Discount.NoDiscount)
        {
            Name = name;
            Barcode = barcode;
            Price = price;
            Discount = discount;
        }
    }
}
